Sources/Core/Shared/Operation.swift

```diff
@@ -303,10 +334,14 @@ public class Operation: NSOperation {
      - parameter errors: an `[ErrorType]` defaults to empty array.
      */
     public func cancelWithErrors(errors: [ErrorType] = []) {
-        if !errors.isEmpty {
-            log.warning("Did cancel with errors: \(errors).")
+        stateLock.withCriticalScope {
+            assert(state != .Finishing, "Operation cannot be cancelled after while it is finishing.")
+            assert(state != .Finished, "Operation cannot be cancelled after it has already finished.")
```

Comment:
If we have thread-safety in `Operation` around cancelling and finishing, are these asserts really necessary now? Their use puts the burden onto the framework consumer to then check that an operation is not getting cancelled at the same time that its finishing - which I guess seems redundant if the base class can work around it. No? Thoughts?